src/core/server.h

```diff
@@ -88,22 +88,27 @@ class InferenceServer {
   Status UnloadModel(const std::string& model_name);
 
   // Register the corresponding shared memory region. Re-register the shared
-  // memory region has been registered.
+  // memory region if it has already been registered.
   Status RegisterSharedMemory(
       const std::string& name, const std::string& shm_key, const size_t offset,
-      const size_t byte_size);
+      const size_t byte_size, const int kind, const int device_id);
 
   // Unregister the corresponding shared memory region.
   Status UnregisterSharedMemory(const std::string& name);
 
   // Unregister all active shared memory regions.
   Status UnregisterAllSharedMemory();
 
-  // Get the address at 'offset' within a shared memory region
+  // Get the address at 'offset' within a system shared memory region
   Status SharedMemoryAddress(
       const std::string& name, size_t offset, size_t byte_size,
       void** shm_mapped_addr);
 
+  // Get the address at 'offset' and byte size of a CUDA shared memory region
+  Status CudaSharedMemoryAddress(
```

Comment:
Why is a different function needed here? Why are we returning a byte size?

Reply:
The byte size of the CUDA shared memory region is stored inside the IPC handle. This method returns the byte_size stored in that handle.

Reply:
But we don't do that for system shared memory.

Reply:
Because the size is already provided in the request headers